src/devices/tfa_14_1504.c

```diff
@@ -0,0 +1,132 @@
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+/**
+Decoder for TFA Dostmann 14.1504 (TPX305)
+
+Payload format:
+- Preamble         {28} 0x7aaaaaa
+- Constant or ID ? {8}  0x5c
+- Type             {4}  (0x2=temperature, 0x7=resync)
+- Temperature      {12} Raw temperature value. Temperature in ºC = (value/4)-532. Example: 0x8a0 = 20 ºC
+- Separator        {8}  0xff (differs if resync)
+- Digest           {16} 16-bit LFSR digest + final XOR
+
+To get raw data:
+```
+rtl_433 -R 0 -X 'n=TFA-14.1504,m=FSK_PCM,s=360,l=360,r=4096,preamble={28}7AAAAAA'
+```
+
+Example payloads (excluding preamble):
+- Resync   = 5c7052f9cee3 (encoding differs from temperature readings => not handled)
+- No probe = 5c2700ffb791 (just like a temperature reading => in this case we report the appropriate probe status and no temperature reading)
+- ...
+- 20 ºC    = 5c28a0ffce69
+- 21 ºC    = 5c28a4ffa0f5
+- ...
+- 24 ºC    = 5c28b0ff6438
+- ...
+- 44 ºC    = 5c2900ff8c9d
+- ...
+*/
+
+#include "decoder.h"
+
+#define NUM_BITS_PREAMBLE 28
+#define NUM_BYTES_DATA    6
+#define OFFSET_MIC        (NUM_BYTES_DATA - 2)
+#define NUM_BITS_DATA     (NUM_BYTES_DATA * 8)
+#define NUM_BITS_TOTAL    (NUM_BITS_PREAMBLE + NUM_BITS_DATA)
+#define NUM_BITS_MAX      (NUM_BITS_TOTAL + 12)
+
+static int tfa_14_1504_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble[] = {
+            0x7a,
+            0xaa,
+            0xaa,
+            0xa0,
+    };
+
+    if (bitbuffer->num_rows != 1) {
+        return DECODE_ABORT_EARLY;
+    }
+    unsigned const row = 0;
+
+    // Optional optimization: Early exit if row too short
+    if (bitbuffer->bits_per_row[row] < NUM_BITS_TOTAL) {
+        return DECODE_ABORT_EARLY; // considered "early" because preamble not checked
+    }
+
+    // Sync on preamble. Typical values: 0, 1
+    unsigned start_pos = bitbuffer_search(bitbuffer, row, 0, preamble, NUM_BITS_PREAMBLE);
+
+    if (start_pos == bitbuffer->bits_per_row[row]) {
+        return DECODE_ABORT_EARLY; // no preamble detected
+    }
+
+    const unsigned available_bits = bitbuffer->bits_per_row[row] - start_pos;
+
+    // check min length
+    if (available_bits < NUM_BITS_TOTAL ||
+            available_bits > NUM_BITS_MAX) {
+        return DECODE_ABORT_LENGTH;
+    }
+
+    uint8_t data[NUM_BYTES_DATA];
+    bitbuffer_extract_bytes(bitbuffer, row, start_pos + NUM_BITS_PREAMBLE, data, NUM_BITS_DATA);
+
+    // Note: Not sure if "5c" is a constant or an ID
+    if (data[0] != 0x5c || data[1] >> 4 != 0x2 || data[3] != 0xff) {
+        return DECODE_FAIL_SANITY;
+    }
+
+    uint16_t calc_mic = lfsr_digest16(data, OFFSET_MIC, 0x8810, 0x93ad) ^ 0xd17c;
+    uint16_t data_mic = (data[OFFSET_MIC] << 8) + data[OFFSET_MIC+1];
+    if (calc_mic != data_mic) {
+        return DECODE_FAIL_MIC;
+    }
+
+    uint16_t raw_temp_c = ((data[1] & 0xf) << 6) + (data[2] >> 2);
```

Comment:
Are you shifting off the last 2 bits to avoid a div 4? Are those never used, i.e. is the precission not 1/4 deg C?

Reply:
> Are you shifting off the last 2 bits to avoid a div 4? Are those never used, i.e. is the precission not 1/4 deg C?

Yes exactly: for each degree C, the value increases by 4. The two lower bits are always zero, as far as I've seen (and it's also seen in the samples.txt file). Also the display on the device itself does not have a fractional part.
For context, it's a barbecue thermometer with a high temperature probe which goes up to 300 degrees C, but is probably not precise enough to justify having a fractional part...

Reply:
Ok, good. Add a `// stripping off last 2 bit as those are always zero` or such ;)  But a float div 4 shoould not be expensive either, as it likely only subs 2 from the exponent, no mantissa manipulation needed.

Reply:
Done!